R/test-that.R

```diff
@@ -124,19 +124,11 @@ test_code <- function(test, code, env, reporter, skip_on_empty = TRUE) {
     # to be able to debug with the DebugReporter
     register_expectation(e, 2)
 
-    e[["handled"]] <- TRUE
     test_error <<- e
+    invokeRestart("end_test")
   }
   handle_fatal <- function(e) {
     handled <<- TRUE
-    # Error caught in handle_error() has precedence
```

Comment:
No longer needed because we jump out of both the tryCatch + withCallingHandlers stack with a restart.